providers/pool.rb

```diff
@@ -120,120 +120,145 @@ def site_identifier
+    is_new_smp_processor_affinity_mask = is_new_value?(doc.root, "APPPOOL/add/cpu/@smpProcessorAffinityMask", @new_resource.smp_processor_affinity_mask.to_s) 
+    is_new_smp_processor_affinity_mask_2 = is_new_value?(doc.root, "APPPOOL/add/cpu/@smpProcessorAffinityMask2", @new_resource.smp_processor_affinity_mask_2.to_s) 
+
+    # Application Pool Config
+    @cmd = "#{appcmd(node)} set config /section:applicationPools"
+
+    # root items
+    configure_application_pool(is_new_auto_start, "autoStart:#{@new_resource.auto_start}")
+    configure_application_pool(is_new_start_mode, "startMode:#{@new_resource.start_mode}")
+    configure_application_pool(@new_resource.runtime_version && is_new_managed_runtime_version, "managedRuntimeVersion:v#{@new_resource.runtime_version}")
+    configure_application_pool(@new_resource.thirty_two_bit && is_new_enable_32_bit_app_on_win_64, "enable32BitAppOnWin64:#{@new_resource.thirty_two_bit}")
+    configure_application_pool(@new_resource.queue_length && is_new_queue_length, "queueLength:#{@new_resource.queue_length}")
+
+    # processModel items
+    configure_application_pool(@new_resource.max_proc && is_new_max_processes, "processModel.maxProcesses:#{@new_resource.max_proc}")
+    configure_application_pool(is_new_load_user_profile, "processModel.loadUserProfile:#{@new_resource.load_user_profile}")
+    configure_application_pool(is_new_set_profile_environment, "processModel.setProfileEnvironment:#{@new_resource.set_profile_environment}")
+    configure_application_pool(is_new_logon_type, "processModel.logonType:#{@new_resource.logon_type}")
+    configure_application_pool(is_new_manual_group_membership, "processModel.manualGroupMembership:#{@new_resource.manual_group_membership}")
+    configure_application_pool(is_new_idle_timeout, "processModel.idleTimeout:#{@new_resource.idle_timeout}")
+    configure_application_pool(is_new_shutdown_time_limit, "processModel.shutdownTimeLimit:#{@new_resource.shutdown_time_limit}")
+    configure_application_pool(is_new_startup_time_limit, "processModel.startupTimeLimit:#{@new_resource.startup_time_limit}")
+    configure_application_pool(is_new_pinging_enabled, "processModel.pingingEnabled:#{@new_resource.pinging_enabled}")
+    configure_application_pool(is_new_ping_interval, "processModel.pingInterval:#{@new_resource.ping_interval}")
+    configure_application_pool(is_new_ping_response_time, "processModel.pingResponseTime:#{@new_resource.ping_response_time}")
+    
+    # recycling items
+    configure_application_pool(@new_resource.recycle_after_time && is_new_recycle_after_time, "recycling.periodicRestart.time:#{@new_resource.recycle_after_time}")
+    configure_application_pool(@new_resource.recycle_at_time && is_new_recycle_at_time, "recycling.periodicRestart.schedule", '-')
+    configure_application_pool(@new_resource.recycle_at_time && is_new_recycle_at_time, "recycling.periodicRestart.schedule.[value='#{@new_resource.recycle_at_time}']")
```

Comment:
I think this is the only command that should have the "+".  Schedule is a collection, so you "+" and "-" to it.  While everything else is a property you are changing and should not have either.  So your add_remove condition maybe should default to blank, then you pass either "+" and "-" when needed for the above two commands.